pkg/webhook/v1beta1/pod/inject_webhook.go

```diff
@@ -170,6 +173,14 @@ func (s *SidecarInjector) Mutate(pod *v1.Pod, namespace string) (*v1.Pod, error)
 	}
 	if needWrapWorkerContainer(trial.Spec.MetricsCollector) {
 		if err = wrapWorkerContainer(trial, mutatedPod, namespace, mountPath, pathKind); err != nil {
+			msg := fmt.Sprintf("Unable to find primary container %v in mutated pod containers %v",
+				trial.Spec.PrimaryContainerName, pod.Spec.Containers)
+			trial.MarkTrialStatusFailed(trialcontroller.TrialFailedReason, msg)
+
+			if err := s.client.Status().Update(context.TODO(), trial); err != nil {
```

Comment:
@johnugeorge Does it sound good to you that we are updating Trial CR status from the Webhook and not from Controller reconcile ?

Reply:
Personally, I do not think it is a good practice. @anencore94 Is there any other possible solution?

Reply:
how about update trial status at `reconcileTrial()` ?
I'm not sure, but somewhere nearby https://github.com/anencore94/katib/blob/7f7a468692a45674852509a7369d15f9c273c012/pkg/controller.v1beta1/trial/trial_controller.go#L215

Reply:
How about sending GET Job API and check if desiredJob is in `Running status`, right after [Create Job](https://github.com/anencore94/katib/blob/7f7a468692a45674852509a7369d15f9c273c012/pkg/controller.v1beta1/trial/trial_controller.go#L302) or [here](https://github.com/anencore94/katib/blob/7f7a468692a45674852509a7369d15f9c273c012/pkg/controller.v1beta1/trial/trial_controller.go#L314)?

Only checking Creation API succeeded is not safe, Is it better if we should check if its status is not being stuck in `Pending` or `PodInitializing` or `ContainerCreating` ?

Reply:
But how we know that status is stuck ? Some pods might need more time to get running.
Is there any way to get Mutation Webhook results in the Controller ?

Reply:
>Some pods might need more time to get running.

I missed this case. I'll check some better options

Reply:
I'm afraid I couldn't find any better idea.

Reply:
Let's discuss this feat in the next community meeting ?
WDYT @anencore94 @gaocegege @johnugeorge ?

Reply:
I'm so sorry but I couldn't join the next meeting. Bi-weekly US friendly meeting schedule is a bit late at my timezone, so I'm afraid I couldn't join when I have a  schedule for the next day.

Could you just share the discussion ? or How about the next-next meeting if this feature is not highly-priored?

Reply:
Sure, that sounds good.

Reply:
Is there any updates at the last meeting ? @andreyvelich

Reply:
We decided to make more research on that and discuss in the next APAC friendly meeting.
cc @johnugeorge

Reply:
Thanks for sharing :)